src/main/java/seedu/address/logic/commands/AddCommand.java

```diff
@@ -56,6 +57,18 @@ public AddCommand(ModuleCode moduleCode, Year year, Semester semester, Grade gra
         this.grade = grade;
     }
 
+    /**
+     * For Testing purposes only
+     * @param validModule
+     */
+    public AddCommand(Module validModule) {
+        requireNonNull(validModule);
+        this.moduleCode = validModule.getModuleCode();
+        this.year = validModule.getYearTaken();
+        this.semester = validModule.getSemesterTaken();
+        this.grade = validModule.getGrade();
+    }
+
```

Comment:
Don't think it is good design to add constructors for testing purposes. The original constructor above should be sufficient for any tests.

Reply:
For situations such as this should I just split into parts then?
![image](https://github.com/AY2324S1-CS2103T-T13-0/tp/assets/108658325/f1c2d241-9f57-4dc1-b16d-9353caa0f8b8)